### PR TITLE
create cat class which extends animal class

### PR DESCRIPTION
**Pull request by codrex**
### ​What​ ​does​ ​this​ ​PR​ ​​do​?
create a class of Cat which is a sub class of Animal

### CODE REVIEW

File: src/cat.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Animal from './main';
+
+export default class Cat extends Animal{
+  constructor(name,legs){
+    super(name,legs);
+  }
+  makeSound(){
+    return 'meows from cat'
+  }
+}
```
Comment (andela-nrotimi) [inline src/cat.js]: New line expected

File: src/cat.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import Animal from './main';
+
+export default class Cat extends Animal{
+  constructor(name,legs){
+    super(name,legs);
```
Comment (andela-nrotimi) [inline src/cat.js]: What does super does?

File: src/main.js [hunk 3]
```diff
@@ -1,12 +1,15 @@
 export default class Animal{
-  constructor(name,legs){
+  constructor(name,legs,sound = 'Animal make sound'){
```
Comment (andela-nrotimi) [inline src/main.js]: Inadequate spacing

File: src/main.js [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
     const data = {name,legs};
     this.getName = ()=> data.name;
     this.getLegs = ()=> data.legs;
     this.setName = name=>data.name = name;
```
Comment (andela-nrotimi) [inline src/main.js]: Inadequate spacing between operators and operands

File: test/test.test.js [hunk 5]
```diff
@@ -18,4 +20,23 @@ describe('Testing Animal class function',()=>{
...
+  it('Encapsulation:: should return Cat',()=>{
+    expect(cat.getName()).to.equal('Cat');
+  })
+  it('Encapsulation:: should return 4',()=>{
+    expect(cat.getLegs()).to.equal('4');
+  })
+  it('Encapsulation:: should return undefined',()=>{
+    expect(cat.legs).to.equal(undefined);
+  })
+  it('Polymorphism: should return meows from cat',()=>{
```
Comment (andela-nrotimi) [inline test/test.test.js]: Does this captures polymorphism?

File: src/cat.js (hunk 1 above)
Comment (codrex) [reply src/cat.js]: ok

File: src/cat.js (hunk 2 above)
Comment (codrex) [reply src/cat.js]: super initialize a parent constructor. it is a reference to the parent class

File: src/main.js (hunk 3 above)
Comment (codrex) [reply src/main.js]: ok

File: src/main.js (hunk 4 above)
Comment (codrex) [reply src/main.js]: ok

File: test/test.test.js (hunk 5 above)
Comment (codrex) [reply test/test.test.js]: yes..

**Pull request merged**